azure-iot-device/azure/iot/device/common/transport/mqtt/mqtt_transport.py

```diff
@@ -5,138 +5,17 @@
-    provider without going through the state machine.
-
-4. When the provider completes an action or receives an acknowledgement, it will call back into this object using
-    functions which are all prefixed with "_on_provider_", such as "_on_provider_connect_complete".  These callback functions
-    will, most of the time, trigger additional state machine transitions by calling _trig_ functions.
-
-5. Functions which are called by the state machine as side-effects of state transitions will accept event_data objects
-    as the second parameter (after self).  The event_data structure contains information about the trigger or the transition
-    which caused the side-effect.
-
-6. Callbacks from this object into caller code, when not passed in as `callback` parameters to function calls, are prefixed with
-    on_ (with no underscore), such as "on_transport_connected'.  Because most callbacks are passed in as function parameters,
-    there are very few callbacks like this.
-
-"""
-
-QOS_SEND_MESSAGE = 1
-QOS_SUBSCRIBE = 1
-
-TOPIC_POS_DEVICE = 4
-TOPIC_POS_MODULE = 6
-TOPIC_POS_INPUT_NAME = 5
-
-
-class TransportAction(object):
-    """
-    base class representing various actions that can be taken
-    when the transport is connected.  When the MqttTransport user
-    calls a function that requires the transport to be connected,
-    a TransportAction object is created and added to the action
-    queue.  Then, when the transport is actually connected, it
-    loops through the objects in the action queue and executes them
-    one by one.
-    """
-
-    def __init__(self, callback):
-        self.callback = callback
-
-
-class SendMessageAction(TransportAction):
-    """
-    TransportAction object used to send a telemetry message or an
-    output message
-    """
-
-    def __init__(self, message, callback):
-        TransportAction.__init__(self, callback)
-        self.message = message
-        self.qos = QOS_SEND_MESSAGE
-
-
-class SubscribeAction(TransportAction):
-    """
-    TransportAction object used to subscribe to a specific MQTT topic
-    """
-
-    def __init__(self, topic, callback):
-        TransportAction.__init__(self, callback)
-        self.topic = topic
-        self.qos = QOS_SUBSCRIBE
-
-
-class UnsubscribeAction(TransportAction):
-    """
-    TransportAction object used to unsubscribe from a specific MQTT topic
-    """
-
-    def __init__(self, topic, callback):
-        TransportAction.__init__(self, callback)
-        self.topic = topic
-
-
-class MethodReponseAction(TransportAction):
-    """
-    TransportAction object used to send a method response back to the service.
-    """
-
-    def __init__(self, method_response, callback):
-        TransportAction.__init__(self, callback)
-        self.method_response = method_response
-
 
 class MQTTTransport(AbstractTransport):
     def __init__(self, auth_provider):
```

Comment:
>auth_provider [](start = 23, length = 13)

I think for DPS this needs to be a security client if we are going to use the same transport.
Since the stage is always using UseSkAuthProvider may we should use some other name for this stage and do another isinstance check

Reply:
This transport is going to move into the iothub module since it's very iothub-specific.  DPS will need a different transport object (or whatever we want to call it) that will do the same thing with a DPS-specific pipeline and ops.

---
In reply to: [280164482](https://github.com/Azure/azure-iot-sdk-python-preview/pull/87#discussion_r280164482) [](ancestors = 280164482)